lib/libzfs/libzfs_util.c

```diff
@@ -596,27 +596,47 @@ zfs_strdup(libzfs_handle_t *hdl, const char *str)
  * Convert a number to an appropriately human-readable output.
  */
 void
-zfs_nicenum(uint64_t num, char *buf, size_t buflen)
+__zfs_nicenum(uint64_t num, char *buf, size_t buflen,
```

Comment:
There's enough code reuse between nicenum and nicetime that I think the extra conditionals and lookup tables in __zfs_nicenum() are justified, although I agree it's borderline.  I'd argue that my incarnation of __zfs_nicenum does make it easier for future nicenum formats to be added though.  One day we might want to add a "ZFS_NICENUM_1000", for example.